metaStoreApp/app/controllers/api/v1/medias_controller.rb

```diff
@@ -1,27 +1,302 @@
 module Api
   module V1
     class MediasController < AccountsController
+      require 'csv'
+
+      # displays the medias based on filters
+      api :GET, '/medias/', 'Displays medias based on filters'
+      param :asset_id, String, desc: 'Asset Id'
+      param :title, String, desc: 'Title of the media'
+      param :duration, :number, desc: 'Duration of the media in milliseconds'
+      example %(
+      Filter by asset_id
+      GET api/v1/medias?asset_id=AB100
+      {
+      "status": "SUCCESS",
+      "message": "Loaded medias",
+      "data": [
+      {
+        "asset_id": "AB100",
+        "media_type": "audio",
+        "account_id": 2,
+        "title": null,
+        "duration": null,
+        "location": null,
+        "recorded_time": null,
+        "timecode": null,
+        "created_at": "2020-02-28T09:12:47.000Z",
+        "updated_at": "2020-02-28T09:12:47.000Z"
+      }
+      ]
+      }
+
+      Filter by title
+      GET api/v1/medias/?title=song4
+      {
+      "status": "SUCCESS",
+      "message": "Loaded medias",
+      "data": [
+      {
+        "asset_id": "AA100",
+        "media_type": "audio",
+        "account_id": 1,
+        "title": "song4",
+        "duration": 60,
+        "location": "https://www.youtube.com/watch?v=pPy0GQJLZUM",
+        "recorded_time": "0001-02-20T01:11:08.000Z",
+        "timecode": "00:00:00.60",
+        "created_at": "2020-02-28T09:12:39.000Z",
+        "updated_at": "2020-02-28T09:13:03.000Z"
+      }
+      ]
+      }
+
+      Filter by duration
+      GET api/v1/medias?duration=12090
+      {
+      "status": "SUCCESS",
+      "message": "Loaded medias",
+      "data": [
+      {
+        "asset_id": "Z100",
+        "media_type": "video",
+        "account_id": 1,
+        "title": "song3",
+        "duration": 12000,
+        "location": "https://www.youtube.com/watch?v=pPy0GQJLZUM",
+        "recorded_time": "0001-01-20T11:00:08.000Z",
+        "timecode": "00:00:12:00",
+        "created_at": "2020-02-28T08:56:41.000Z",
+        "updated_at": "2020-02-28T09:07:20.000Z"
+      },
+      {
+        "asset_id": "AA100",
+        "media_type": "audio",
+        "account_id": 1,
+        "title": "song4",
+        "duration": 60,
+        "location": "https://www.youtube.com/watch?v=pPy0GQJLZUM",
+        "recorded_time": "0001-02-20T01:11:08.000Z",
+        "timecode": "00:00:00.60",
+        "created_at": "2020-02-28T09:12:39.000Z",
+        "updated_at": "2020-02-28T09:13:03.000Z"
+      }
+      ]
+      }
+      )
       def index
-        @@medias = Media.all
-        render json: { status: 'SUCCESS', message: 'Loaded medias', data: @@medias }, status: :ok 
+        @medias = Media.where(nil)
```

Comment:
default limit and offset has to be applied
should be able to pass multiple filters together
title and asset_id search need not be an exact match, it should be substring match.
for eg query 'amagi' should match assets 'ASSET_AMAGI_001' and '001AMAGI'